### PR TITLE
Rewrite to upgrade mega to 10.2.4

### PR DESCRIPTION
**Pull request by alexlegresley**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [X] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).
- [X] `brew audit --cask {{cask_file}}` is error-free.
- [X] `brew style --fix {{cask_file}}` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [ ] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [ ] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).
- [ ] `brew audit --new-cask {{cask_file}}` worked successfully.
- [X] `brew install --cask {{cask_file}}` worked successfully.
- [X] `brew uninstall --cask {{cask_file}}` worked successfully.

The distribution structure of Mega changed significantly from version 7.0.26 to 10.2.4, requiring substantial revisions to accommodate.

### CODE REVIEW

File: Casks/mega.rb [hunk 1]
```diff
@@ -1,11 +1,30 @@
 cask "mega" do
```
Comment (vitorgalvao) [inline Casks/mega.rb:1]: Need to be renamed to `megax`.

**Review by vitorgalvao — DISMISSED** [review]

File: Casks/mega.rb (hunk 1 above)
Comment (alexlegresley) [reply Casks/mega.rb]: Made the change renaming the cask to megax. Does this mean the file name must also be changed to megax? Is this a change I can make from my end?

Comment (vitorgalvao) [reply Casks/mega.rb]: > Does this mean the file name must also be changed to megax? Is this a change I can make from my end?

Yes to both.

Comment (alexlegresley) [reply Casks/mega.rb]: Added megax.rb and deleted mega.rb. I know for formula name changes it has to be added to formula_changes.json, but I can't find an equivalent in the documentation for cask. Are there any other steps for renaming casks?

**Comment by miccal** [comment]
Thank you for your contribution to `homebrew-cask` @alexlegresley.

**Review by miccal — APPROVED** [review]

**Pull request merged**